Sources/PackageModel/Manifest/PackageDependencyDescription.swift

```diff
@@ -22,17 +27,21 @@ public enum PackageDependency: Equatable {
     public struct SourceControl: Equatable, Encodable {
         public let identity: PackageIdentity
         public let nameForTargetDependencyResolutionOnly: String?
-        public let location: String
+        public let location: Location
         public let requirement: Requirement
         public let productFilter: ProductFilter
 
-        /// The dependency requirement.
         public enum Requirement: Equatable, Hashable {
             case exact(Version)
             case range(Range<Version>)
             case revision(String)
             case branch(String)
         }
+
+        public enum Location: Equatable {
+            case local(AbsolutePath)
+            case remote(Foundation.URL)
```

Comment:
In the past I think we had some issues with Foundation representing `git@...` URLs, at least on Darwin.  I agree that using URL instead of String here makes more sense conceptually, but do we think there might be side effects?  We should of course report them if there, but this might be something to watch for.

Reply:
this PR helps remove the logic that depends on checking if there is a scheme or not, and instead uses the type: `localSourceControl` vs `remoteSourceControl`, so ideally this should no longer be an issue regardless

Reply:
I see.  Yes, if SwiftPM is no longer asking those same questions but mainly using the URL as a container of URL-formatter strings, then that might go better.

Reply:
yes this is the goal - make the concrete location opaque to logic, so it only gets used when we pass it to git, where we use absoluteString to write out the full value

is there a specific use case you recall that we need to add a test for to make sure this works correctly in that sense?